File: Python/268. Missing Number.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = nums[0] ^ 0
        for i in range(1, len(nums)):
            ans ^= i ^ nums[i]
        return ans ^ len(nums)

File: 2021/268. Missing Number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int ans = nums.size();
        for (int i = 0; i < nums.size(); i++) {
            ans ^= nums[i] ^ i;
        }
        return ans;
    }
};

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int res = 0 ^ nums[0], N = nums.size();
        for (int i = 1; i < nums.size(); i++) {
            res ^= i ^ nums[i];
        }
        return res ^ N;
    }
};